include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,505 @@
+        {
+            return it -= n;
+        }
+
+        friend constexpr decltype(auto) iter_move(const iterator& x) noexcept(
+            (noexcept(std::ranges::iter_move(std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>())) && ...) &&
+            (std::is_nothrow_move_constructible_v<std::ranges::range_rvalue_reference_t<__maybe_const<Const, Views>>> && ...))
+        {
+            auto __tr = [](auto&&... __args) -> decltype(auto) {
+                return rvalue_reference_type(std::forward<decltype(__args)>(__args)...);
+            };
+            return apply_to_tuple(__tr, std::ranges::iter_move, x.current_);
+
+        }
+
+        friend constexpr void iter_swap(const iterator& x, const iterator& y) noexcept(
+            (noexcept(std::ranges::iter_swap(std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>(),
+                                        std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>())) && ...))
+          requires(std::indirectly_swappable<std::ranges::iterator_t<__maybe_const<Const, Views>>> && ...)
+        {
+            bi_apply_to_tuple(std::ranges::iter_swap, x.current_, y.current_);
+        }
+
+      private:
+        template <std::size_t... In>
+        constexpr bool
+        compare_equal(iterator y, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(y.current_)) || ...);
+        }
+
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr bool
+        compare_with_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(sentinels)) || ...);
+        }
+        
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>        
+        distance_to_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return std::ranges::min({difference_type(std::get<In>(current_) - std::get<In>(sentinels))...}, std::less{},
+                                    [](auto a){ return std::abs(a);});
+        }
+        
+        template <std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>
+        distance_to_it(const iterator it, std::index_sequence<In...>) const
+        {
+            return std::ranges::min({difference_type(std::get<In>(it.current_) - std::get<In>(current_))...}, std::less{},
+                                    [](auto a){ return std::abs(a);});
+        }
+
+        friend class zip_view;
+
+        iterator_type current_;
+    }; // class iterator
+
+    template <bool Const>
+    class sentinel
+    {
+        template <typename... Sentinels>
+        constexpr explicit sentinel(const Sentinels&... sentinels) : end_(sentinels...)
```

Comment:
Lets align with iterator:

```suggestion
        constexpr explicit sentinel(Sentinels... sentinels) : end_(std::move(sentinels)...)
```